src/picongpu/include/particles/synchrotronPhotons/PhotonCreator.hpp

```diff
@@ -0,0 +1,289 @@
+
+            return DELTA_T * charge*charge * mass*SPEED_OF_LIGHT / (float_X(4.0)*PI*EPS0 * HBAR*HBAR) *
+                float_X(1.732050807) / (float_X(2.0) * PI) * chi/gamma * (float_X(1.0) - delta) / delta *
+                (this->curF_1[z] + float_X(1.5) * delta * chi * z * this->curF_2[z]);
+        }
+        else
+        {
+            // classical
+            const float_X z = float_X(2.0/3.0) * delta / chi;
+
+            return DELTA_T * charge*charge * mass*SPEED_OF_LIGHT / (float_X(4.0)*PI*EPS0 * HBAR*HBAR) *
+                float_X(1.732050807) / (float_X(2.0) * PI) * chi/gamma / delta * this->curF_1[z];
+        }
+    }
+
+    DINLINE float_X emission_prob_modified(
+        const float_X z,
+        const float_X chi,
+        const float_X gamma,
+        const float_X mass,
+        const float_X charge) const
+    {
+        return float_X(3.0) * z*z * emission_prob(
+            z*z*z,
+            chi, gamma, mass, charge);
+    }
+
+    /** Return the number of target particles to be created from each source particle.
+     *
+     * Called for each frame of the source species.
+     *
+     * @param sourceFrame Frame of the source species
+     * @param localIdx Index of the source particle within frame
+     * @return number of particle to be created from each source particle
+     */
+    DINLINE unsigned int numNewParticles(FrameType& sourceFrame, int localIdx)
+    {
+        using namespace PMacc::algorithms;
+
+        PMACC_AUTO(particle, sourceFrame[localIdx]);
+
+        ValueType_E fieldE;
+        ValueType_B fieldB;
+
+        /* particle position, used for field-to-particle interpolation */
+        floatD_X pos = particle[position_];
+        const int particleCellIdx = particle[localCellIdx_];
+        /* multi-dim coordinate of the local cell inside the super cell */
+        DataSpace<TVec::dim> localCell(DataSpaceOperations<TVec::dim>::template map<TVec > (particleCellIdx));
+        /* interpolation of E- */
+        const fieldSolver::numericalCellType::traits::FieldPosition<FieldE> fieldPosE;
+        fieldE = Field2ParticleInterpolation()
+            (cachedE.shift(localCell).toCursor(), pos, fieldPosE());
+        /*                     and B-field on the particle position */
+        const fieldSolver::numericalCellType::traits::FieldPosition<FieldB> fieldPosB;
+        fieldB = Field2ParticleInterpolation()
+            (cachedB.shift(localCell).toCursor(), pos, fieldPosB());
+
+        const float3_X mom = particle[momentum_] / particle[weighting_];
+        const float_X mom2 = math::dot(mom, mom);
+        const float3_X mom_norm = mom * math::rsqrt(mom2);
+        const float_X mass = frame::getMass<FrameType>();
+        const float_X mc = mass * SPEED_OF_LIGHT;
+
+        const float_X gamma = math::sqrt(float_X(1.0) + mom2 / (mc*mc));
+        const float3_X vel = mom / (gamma * mass); // low accuracy?
```

Comment:
1. There is no speed of light in this formula
2. I really dont need precision here because the whole algorithm assumes highly relativistic electrons.